### PR TITLE
faster avatar

### PR DESCRIPTION
**Pull request by chrisnojima**
Some speedups for avatar
- [x] skip loading call if we aren't using a placeholder
- [x] inline some classes so we can skip mounting
- [x] rAF the load call

### CODE REVIEW

File: shared/common-adapters/avatar.js [hunk 1]
```diff
@@ -186,27 +184,28 @@ const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps) => {
...
-  ])
+  const style = Styles.isMobile
+    ? ownProps.style
+    : Styles.collapseStyles([
+        ownProps.style,
+        onClick && Styles.platformStyles({isElectron: Styles.desktopStyles.clickable}),
+      ])
 
   let url = stateProps._urlMap ? urlsToImgSet(stateProps._urlMap.toObject(), ownProps.size) : null
+  let load
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.js]: only load if we're showing a placeholder

File: shared/common-adapters/avatar.js [hunk 2]
```diff
@@ -231,44 +228,12 @@ const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps) => {
   }
 }
 
-class AvatarConnector extends React.PureComponent<Props> {
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.js]: remove this extra class that just does the loading logic

File: shared/common-adapters/avatar.render.desktop.js [hunk 3]
```diff
@@ -5,6 +5,14 @@ import * as Styles from '../styles'
 import type {Props} from './avatar.render'
 
 class AvatarRender extends React.PureComponent<Props> {
+  componentDidUpdate(prevProps: Props) {
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.render.desktop.js]: removed from shim class above. now use rAF

File: shared/common-adapters/avatar.render.native.js [hunk 4]
```diff
@@ -32,77 +29,63 @@ const sizeToTeamBorderRadius = {
 // Android doesn't handle background colors border radius setting
 const backgroundOffset = 1
 
-class Background extends React.PureComponent<{loaded: boolean, loadingColor: any, borderRadius: number}> {
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.render.native.js]: removing extra classes and just using a helper function

File: shared/common-adapters/avatar.render.native.js [hunk 5]
```diff
@@ -32,77 +29,63 @@ const sizeToTeamBorderRadius = {
...
-}
+const border = ({borderColor, borderRadius}) => (
+  <Kb.Box style={[styles.borderBase, {borderColor, borderRadius}]} />
+)
 
 class AvatarRender extends React.PureComponent<Props, State> {
-  state: State = {loaded: false}
+  constructor(props: Props) {
+    super(props)
+    this.state = {loaded: !!this.props.load}
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.render.native.js]: only load if we need to load

**Comment by chrisnojima** [comment]
@keybase/react-hackers speeds up avatar a little

File: shared/common-adapters/avatar.render.native.js [hunk 6]
```diff
@@ -32,77 +29,63 @@ const sizeToTeamBorderRadius = {
...
-    return (
-      <NativeImage
-        source={this.props.url}
-        onLoadEnd={this.props.onLoadEnd}
-        style={[styles[`image:${this.props.size}`], {borderRadius, opacity}]}
-      />
-    )
-  }
-}
+const background = props => (
```
Comment (buoyad) [inline shared/common-adapters/avatar.render.native.js]: Seems like `PureComponent` would render less than this, do you want to use `React.memo` for these?

File: shared/common-adapters/avatar.render.native.js [hunk 7]
```diff
@@ -112,44 +95,42 @@ class AvatarRender extends React.PureComponent<Props, State> {
...
-            <Border
-              borderColor={this.props.borderColor || globalColors.black_10}
-              borderRadius={borderRadius}
-            />
-          )}
+      <Kb.ClickableBox onClick={this.props.onClick} feedback={false} style={containerStyle}>
+        <Kb.Box style={containerStyle}>
+          {!this.props.skipBackground &&
+            (!this.props.skipBackgroundAfterLoaded || !this.state.loaded) &&
+            background({
```
Comment (buoyad) [inline shared/common-adapters/avatar.render.native.js]: Capitalize and treat like components

Comment (chrisnojima) [reply shared/common-adapters/avatar.render.native.js]: These are not components on purpose

Comment (buoyad) [reply shared/common-adapters/avatar.render.native.js]: Ooooh ok TIL the difference. Why would we ever use jsx over this for functional components? I guess for hooks, now. Well anyway sounds good.

Comment (chrisnojima) [reply shared/common-adapters/avatar.render.native.js]: this isn't really a sub component where some of its props might change. so making it a real component like it used to do just creates more overhead. if you look at desktop its all just one function. this means if the parent renders this does too but thats what happens anyways practically and its a lot faster to just call it and not have this also mount

File: shared/common-adapters/avatar.js [hunk 8]
```diff
@@ -112,37 +110,42 @@ class SharedAskForUserData {
...
+          this._teamLastReq[k] = now
+          return true
+        }
+        return false
+      })
+      this._teamQueue = {}
+      this._userQueue = {}
+      if (usernames.length || teamnames.length) {
+        requestAnimationFrame(() => {
+          console.log('aaa making dispatch for avatars', usernames)
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/avatar.js]: done

**Pull request merged**